Removed links to articles with less information than this article.

ceford opened the pull request:
Also fixed W2W: front-end to frontend and back-end to backend.

HLeithner commented:
iot makes no sense to remove that links, since they are linking to the section (intro page) with all information to plugins or templates. Better move the information from this page to the other pages. or provide more information at the intro/landing pages

ceford commented:
The links I removed were examples of poor practice - perhaps a matter of opinion but I could provide some evidence of that. So perhaps we need some better documentation guidelines. There are such for the Wiki.

In this case I have added back specific links below each block to say what they are for. I took out the lines that say *Management feature: * because they have little meaning (to me) and I don't think they add anything useful here.

ceford commented:
The links I removed were examples of poor practice - perhaps a matter of opinion but I could provide some evidence of that. So perhaps we need some better documentation guidelines. There are such for the Wiki.

In this case I have added back specific links below each block to say what they are for. I took out the lines that say *Management feature: * because they have little meaning (to me) and I don't think they add anything useful here.

The original document has a link below its Component information block: https://docs.joomla.org/Extension_types_(general_definitions)

HLeithner commented:
I'm not sure how a "read here the full story" link can be a poor practice, ymmv, the variant you added now seems to be nicer of course. I merge this for now thanks
